control-plane-operator/main.go

```diff
@@ -126,14 +132,24 @@ func NewStartCommand() *cobra.Command {
 			RetryPeriod:                   &retryPeriod,
 			HealthProbeBindAddress:        healthProbeAddr,
 			NewCache: cache.BuilderWithOptions(cache.Options{
-				DefaultSelector:   cache.ObjectSelector{Field: fields.OneTermEqualSelector("metadata.namespace", namespace)},
-				SelectorsByObject: cache.SelectorsByObject{&operatorv1.IngressController{}: {Field: fields.OneTermEqualSelector("metadata.namespace", manifests.IngressPrivateIngressController("").Namespace)}},
+				DefaultSelector: cache.ObjectSelector{Field: fields.OneTermEqualSelector("metadata.namespace", namespace)},
+				SelectorsByObject: cache.SelectorsByObject{
+					&operatorv1.IngressController{}: {Field: fields.OneTermEqualSelector("metadata.namespace", manifests.IngressPrivateIngressController("").Namespace)},
+					&corev1.Event{}:                 {Field: fields.OneTermEqualSelector("type", "warning")},
```

Comment:
nit: May be add a comment
`// We watch warning events to be able to surface cloud provider errors as conditions`.
 
Ideally we would reference an issue tracking lack of conditions in cloud providers upstream. If there's nothing opened may be create one with one particular example we are seeing on events but not in conditions to get the discussion going.